game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -118,34 +125,48 @@ def skynet_nodes(self, game: Game) -> list[SkynetNode]:
             if game.iads_considerate_culling(node.group.ground_object):
                 # Skip culled ground objects
                 continue
-            try:
-                skynet_node = SkynetNode.from_group(node.group)
-                for connection in node.connections.values():
-                    if connection.ground_object.is_friendly(
-                        skynet_node.player
-                    ) and not game.iads_considerate_culling(connection.ground_object):
-                        skynet_node.connections[connection.iads_role.value].append(
-                            SkynetNode.dcs_name_for_group(connection)
-                        )
-                skynet_nodes.append(skynet_node)
-            except IadsNetworkException:
-                # Node not skynet compatible
-                continue
+
+            skynet_node = SkynetNode.from_group(node.group)
+            for connection in node.connections.values():
+                is_friendly = connection.ground_object.is_friendly(skynet_node.player)
+                is_culled = game.iads_considerate_culling(connection.ground_object)
+                if is_friendly and not is_culled:
+                    key = connection.iads_role.value
+                    name = SkynetNode.dcs_name_for_group(connection)
+                    skynet_node.connections[key].append(name)
+            skynet_nodes.append(skynet_node)
+
         return skynet_nodes
 
+    def update_iads_building(self, tgo: TheaterGroundObject) -> None:
+        if not isinstance(tgo, IadsBuildingGroundObject):
+            return
+
+        from game.server import EventStream
+
+        for node in self.nodes:
+            for cID, group in node.connections.items():
+                if group.ground_object is tgo:
+                    self._push_deleted_connection(cID)
+        # if tgo.alive_unit_count > 0:
+        self._update_network(tgo)
+
     def update_tgo(self, tgo: TheaterGroundObject) -> None:
         """Update the IADS Network for the given TGO"""
         # Remove existing nodes for the given tgo
         for cn in self.nodes:
             if cn.group.ground_object == tgo:
                 self.nodes.remove(cn)
-        try:
-            # Create a new node for the tgo
-            self.node_for_tgo(tgo)
-            # TODO Add the connections or calculate them..
-        except IadsNetworkException:
+                for cID in cn.connections:
+                    self._push_deleted_connection(cID)
+        # Create a new node for the tgo
+        # TODO Add the connections or calculate them..
+        node = self.node_for_tgo(tgo)
+        if node is None:
             # Not participating
-            pass
+            return
+        self.nodes.append(node)
+        self._connect_to_network(node, tgo)
```

Comment:
Replacing the exception, line 169 is also new, it was missing functionality for connecting the new node to the network.